### PR TITLE
Add support for Content Alignment inside AutoSuggestBox

### LINKED ISSUE #3284: Autosugest box unaligned text
**Describe the bug**
The autosugest box's placeholder and content text do not center-align vertically when indicated. Changing it to bottom doesn't appears to work either.

**Steps to reproduce the bug**

1- Create a Autosugest box using the following boilerplate code:

```
	<Grid x:Name="ContentArea" Margin="{StaticResource MediumLeftRightMargin}">
		<Grid.RowDefinitions>
			<RowDefinition Height="Auto" />
			<RowDefinition Height="*" />
		</Grid.RowDefinitions>
		<!-- Top Bar Definitions -->
		<Grid Grid.Row="0" Margin="15">
			<Grid.ColumnDefinitions>
				<ColumnDefinition Width="2.5*" />
				<ColumnDefinition Width="1.5*" />
			</Grid.ColumnDefinitions>
			<!-- Search Box -->
			<StackPanel Grid.Column="0" Orientation="Horizontal" VerticalAlignment="Center">
				<AutoSuggestBox x:Name="SearchDownload" Width="475" Height="45"
				PlaceholderText="Find a downloaded item" QueryIcon="Find"
				VerticalContentAlignment="Bottom" HorizontalContentAlignment="Left"
				TextChanged="SearchDownload_TextChanged" QuerySubmitted="SearchDownload_QuerySubmitted" SuggestionChosen="SearchDownload_SuggestionChosen" />
			</StackPanel>
         </Grid>

```

**Expected behavior**
The text should modify its centering relative to the control itself.

**Screenshots**

Autosugest box without content
Autosugest box with content

**Version Info**
2.4.3 recently updated but it didn't solve it either. Used an older version (I think it was 2.2.1) and the problem existed there too, that's why I updated the dependence.

NuGet package version: 
Microsoft.UI.Xaml 2.4.3

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes |  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       | Yes   |
| May 2019 Update (18362)            | Yes  |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

### PR DESCRIPTION
**Pull request by beervoley**
## Description
This PR adds binding of `VerticalContentAlignment` and `HorizontalContentAlignment` to AutoSuggestBox.

## Motivation and Context

This PR fixes #3284 
Before, it was impossible to set the content alignment inside AutoSuggestBox using the appropriate arguments in the constructor. This PR makes it possible to set content alignment to the desired values via:

`<AutoSuggestBox VerticalContentAlignment="Bottom" HorizontalContentAlignment="Left"/>`

If the user doesn't provide the content alignment values, the content will be in the left top corner of the box.
## How Has This Been Tested?

Manually tested the changes (see the attached screenshot).

## Screenshots (if appropriate):

Content position without providing any content alignment properties:
image

Content position with `VerticalContentAlignment = "Bottom"` and `HorizontalContentAlignment = "Center"`:
image

### CODE REVIEW

File: dev/AutoSuggestBox/AutoSuggestBox_themeresources.xaml [hunk 1]
```diff
@@ -315,6 +315,8 @@
                             IsHorizontalRailEnabled="{TemplateBinding ScrollViewer.IsHorizontalRailEnabled}"
                             IsVerticalRailEnabled="{TemplateBinding ScrollViewer.IsVerticalRailEnabled}"
                             IsDeferredScrollingEnabled="{TemplateBinding ScrollViewer.IsDeferredScrollingEnabled}"
+                            VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
+                            HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
```
Comment (chingucoding) [inline dev/AutoSuggestBox/AutoSuggestBox_themeresources.xaml:319]: I think it makes sense to add those bindings to the regular textbox style too.

Comment (beervoley) [reply dev/AutoSuggestBox/AutoSuggestBox_themeresources.xaml:319]: By regular - you mean? If you mean the actual textbox template code, it's still in the OS :)

Comment (Felix-Dev) [reply dev/AutoSuggestBox/AutoSuggestBox_themeresources.xaml:319]: The default TextBox template has been pulled out of the OS into WinUI: https://github.com/microsoft/microsoft-ui-xaml/blob/626c9be99f6cccf8c59b19cdfa5bd245563762fa/dev/CommonStyles/TextBox_themeresources.xaml#L91

Comment (chingucoding) [reply dev/AutoSuggestBox/AutoSuggestBox_themeresources.xaml:319]: @Felix-Dev beat me to it. Yes the default TextBox template (and RichEditBox template) have been lift out of the OS into WInUI 2.

Comment (beervoley) [reply dev/AutoSuggestBox/AutoSuggestBox_themeresources.xaml:319]: @Felix-Dev thank you for pointing this out :) Didn't know about it! I'll have a look in a few:)

**Comment by beervoley** [comment]
@chingucoding @Felix-Dev added it to the default text box:)

**Comment by StephenLPeters** [comment]
@beervoley You will need to update the visual verification files in order to resolve these test failures, reach out to me if you need help resolving that.

**Comment by StephenLPeters** [comment]
@beervoley can you merge master into this branch? that will pick up the fix for the visual verifications files that went in today then we can rerun the pipeline to get the verification files you need, and rerun with those to get a clear test pass :)

**Review by StephenLPeters — APPROVED** [review]

**Pull request merged**